app/models/user.rb

```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, allow_nil: true
+
```

Comment:
- [ ] The `posts_counter` column should be of data type integer. This means that it should not have `nil` values because `nil` is another data type. Also, setting validations to disallow nil values for a column ensures data integrity and helps prevent potential errors or unexpected behaviour. Kindly remove the validation that allows the `posts_counter` column to store `nil` values. 👍 😎